src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1388,6 +1448,101 @@ export class SavedObjectsRepository {
     });
   }
 
+  /**
+   * Removes one or more workspace from a given saved object. If no workspace remain, the saved object is deleted
+   * entirely. This method and [`addToWorkspaces`]{@link SavedObjectsRepository.addToWorkspaces} are the only ways to change which workspace a
+   * saved object is shared to.
+   */
+  async deleteFromWorkspaces(
```

Comment:
It looks this function has not yet used, I guess we should always call this function when deleting object within a workspace, am I right?

Reply:
It's not used for now. when we should call this function that depends on the UX flow of share. The `delete` method is used for all saved objects deletion and should works well before share is implemented.